.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,105 @@
+name: "test-p4c"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  # Build and test p4c on Ubuntu 16.04
+  build-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        unified: [ON, OFF]
+        enable_gmp: [ON, OFF]
+        exclude:
+          - unified: OFF
+            enable_gmp: OFF
+    runs-on: ubuntu-16.04
+    env:
+      CTEST_PARALLEL_LEVEL: 4
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+
+    - name: Build (Linux)
+      run: |
+        tools/start_ccache
+        # To flush out issues with unified vs. non-unified builds,
+        # do a non-unified build before continuing with the rest,
+        # which produces a unified build.
+        docker build --network ccache_network -t p4c --build-arg IMAGE_TYPE=test --build-arg ENABLE_UNIFIED_COMPILATION=${{matrix.unified}} --build-arg ENABLE_GMP=${{ matrix.enable_gmp }} .
+
+    - name: Run tests (Linux)
+      run: |
+        sudo docker run --privileged -w /p4c/build -e $CTEST_PARALLEL_LEVEL p4c ctest --output-on-failure --schedule-random
```

Comment:
it may be worth it to keep the comment we had in travis.yml to explain why `sudo` and `--privileged` are required for these tests